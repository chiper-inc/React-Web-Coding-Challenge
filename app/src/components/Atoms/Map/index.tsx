import React from 'react';
import { MapContainer, Marker, Popup } from 'react-leaflet';
import 'leaflet/dist/leaflet.css';
import 'leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css';
import 'leaflet-defaulticon-compatibility';

// import { Container } from './styles';

const Map: React.FC = () => {
  return (
    <MapContainer
      center={[40.8054, -74.0241]}
      zoom={14}
      scrollWheelZoom={false}
      style={{ height: `100%`, width: `100%` }}
    >
      <Marker position={[40.8054, -74.0241]} draggable={true}>
        <Popup>Hey ! you found me</Popup>
      </Marker>
    </MapContainer>
  );
};

export default Map;
