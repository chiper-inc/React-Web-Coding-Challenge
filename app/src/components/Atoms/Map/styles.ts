import { globalStyles } from '@/styles/globalStyles';
import styled from 'styled-components';

export const Container = styled.div`
  margin-bottom: ${globalStyles.metrics.margin};
`;
