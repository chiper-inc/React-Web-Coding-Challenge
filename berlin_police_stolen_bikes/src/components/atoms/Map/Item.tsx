import { FC, LegacyRef } from 'react'

interface MapItemProps {
	mapRef: LegacyRef<HTMLDivElement> | undefined
}

const MapItem: FC<MapItemProps> = ({ mapRef }) => {
	return <div ref={mapRef} className={'w-full h-mid-screen'} />
}

export default MapItem
