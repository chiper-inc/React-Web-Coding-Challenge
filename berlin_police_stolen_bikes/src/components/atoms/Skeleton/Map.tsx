import { FiMap } from 'react-icons/fi/'

const SkeletonMap = () => {
	return (
		<figure className={'h-mid-screen w-full flex bg-slate-200 dark:bg-slate-700'}>
			<FiMap
				className={
					'w-4/5 h-auto md:w-auto md:h-4/5 m-auto text-slate-600 dark:text-slate-300 transition-colors ease-in-out' +
					' duration-500 animate-pulse'
				}
			/>
		</figure>
	)
}

export default SkeletonMap
