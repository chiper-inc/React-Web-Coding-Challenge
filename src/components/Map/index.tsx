import { MapType } from '../../types/types';
import ReactMapGl from 'react-map-gl';
import 'mapbox-gl/dist/mapbox-gl.css';

interface MapProps {
  latitude: MapType['latitude'];
  longitude: MapType['longitude'];
  zoom: MapType['zoom'];
}
const mapboxToken =
  'pk.eyJ1IjoiZW1hbnVlbGp0YyIsImEiOiJja3ZqcjZsY2UybzNmMnVxaTVwNXpsMXd5In0.AXN4b5miAcgBzZUBw66sxw';

const Map = ({
  latitude = 40.78343,
  longitude = -73.96625,
  zoom = 11,
}: MapProps) => {
  console.log('props map', latitude, longitude, zoom);
  const withMap = '100%';
  const heightMap = '400px';
  return (
    <div className="w-full">
      <ReactMapGl
        width={withMap}
        height={heightMap}
        latitude={latitude}
        longitude={longitude}
        zoom={zoom}
        //onViewportChange={(viewport) => this.setState({ viewport })}
        mapboxApiAccessToken={mapboxToken}
        mapStyle="mapbox://styles/mapbox/streets-v10"
      />
    </div>
  );
};

export default Map;
