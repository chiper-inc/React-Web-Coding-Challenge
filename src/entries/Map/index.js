import React from 'react'

export const Map = () => <div>Hello World!</div>
