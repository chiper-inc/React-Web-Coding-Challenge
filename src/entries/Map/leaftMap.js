/* eslint-disable camelcase */
import 'leaflet/dist/leaflet.css'
import './assets/leaftMap.css'
import L from 'leaflet'

const noImageSrc =
  'https://upload.wikimedia.org/wikipedia/commons/5/5a/No_image_available_500_x_500.svg'
const URL_MAPPER =
  'https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}?access_token=pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4NXVycTA2emYycXBndHRqcmZ3N3gifQ.rJcFIG214AriISLbB6B5aw'
const DEF_CONFIG = {
  maxZoom: 18,
  attribution:
    'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors, ' +
    '<a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, ' +
    'Imagery Â© <a href="https://www.mapbox.com/">Mapbox</a>',
  id: 'mapbox/streets-v11',
  tileSize: 512,
  zoomOffset: -1
}

const renderContentPopUp = (title, srcImg, occurredDate) => {
  const container = document.createElement('div')
  const titleElem = document.createElement('h6')
  const ocurredElem = document.createElement('p')
  const imgElem = document.createElement('img')

  container.className = 'container-fluid text-center'
  container.id = 'popup-map-stolenbikes'
  titleElem.innerHTML = title || 'Unknown'
  titleElem.className = 'title-popup-stolen-bikes'
  imgElem.src = srcImg || noImageSrc
  imgElem.className = 'thumb-popup-stolen-bikes'
  ocurredElem.innerHTML = `<span class="span-text-popup-stolen-bikes">Ocurred at</span> ${occurredDate}`

  container.appendChild(titleElem)
  container.appendChild(imgElem)
  container.appendChild(ocurredElem)

  return container
}

const renderPopUp = (feature, layer) => {
  const {
    properties: { description, occurred_at, title }
  } = feature
  const dummy = document.createElement('div')
  dummy.innerHTML = description
  const img = dummy.getElementsByTagName('img')[0]?.src
  const content = renderContentPopUp(title, img, occurred_at)
  layer.bindPopup(content)
}

export default class LeaftLetMap {
  constructor (htmlElement, rootView) {
    this.leafletMap = null
    this.htmlElement = htmlElement
    this.rootView = rootView

    this.renderMap = this.renderMap.bind(this)
    this.renderGeoJson = this.renderGeoJson.bind(this)
  }

  renderMap () {
    if (!this.leafletMap) {
      this.leafletMap = L.map(this.htmlElement)
    }
    this.leafletMap.setView(this.rootView, 13)
    L.tileLayer(URL_MAPPER, DEF_CONFIG).addTo(this.leafletMap)
  }

  renderGeoJson (geoJson) {
    L.geoJSON([geoJson], {
      onEachFeature: renderPopUp,
      pointToLayer: (feature, latlng) =>
        L.circleMarker(latlng, {
          radius: 8,
          fillColor: '#007bff',
          color: '#000',
          weight: 1,
          opacity: 1,
          fillOpacity: 0.8
        })
    }).addTo(this.leafletMap)
  }

  DisposeElem () {
    this.leafletMap = null
  }
}
