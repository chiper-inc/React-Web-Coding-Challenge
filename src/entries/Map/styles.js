import styled from 'styled-components/macro'

export const Container = styled.div`
  width: 100%;
  height: 100%;
  position: relative;
  outline: none;
  padding: 1rem 0;
  border: 1px solid #ccc;
`
