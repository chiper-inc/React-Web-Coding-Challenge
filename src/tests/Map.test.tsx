import React from 'react';
import { shallow } from 'enzyme';
import Map from '../components/Map';

test('renders the component Map', () => {
  const component = shallow(
    <Map latitude={7854126} longitude={5414578} zoom={11} />
  );
  expect(component).toMatchSnapshot();
});
