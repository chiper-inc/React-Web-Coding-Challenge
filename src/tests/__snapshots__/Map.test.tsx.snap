// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component Map 1`] = `
<div
  className="w-full"
>
  <ForwardRef
    altitude={1.5}
    asyncRender={false}
    attributionControl={true}
    bearing={0}
    className=""
    clickRadius={0}
    container={<body />}
    disableTokenWarning={false}
    doubleClickZoom={true}
    dragPan={true}
    dragRotate={true}
    eventRecognizerOptions={Object {}}
    getCursor={[Function]}
    height="400px"
    keyboard={true}
    latitude={7854126}
    longitude={5414578}
    mapOptions={Object {}}
    mapStyle="mapbox://styles/mapbox/streets-v10"
    mapboxApiAccessToken="pk.eyJ1IjoiZW1hbnVlbGp0YyIsImEiOiJja3ZqcjZsY2UybzNmMnVxaTVwNXpsMXd5In0.AXN4b5miAcgBzZUBw66sxw"
    mapboxApiUrl="https://api.mapbox.com"
    maxPitch={85}
    maxZoom={24}
    minPitch={0}
    minZoom={0}
    onClick={null}
    onContextMenu={[Function]}
    onError={[Function]}
    onHover={null}
    onLoad={[Function]}
    onNativeClick={null}
    onResize={[Function]}
    onTransitionEnd={[Function]}
    onTransitionInterrupt={[Function]}
    onTransitionStart={[Function]}
    onViewStateChange={null}
    onViewportChange={null}
    pitch={0}
    preserveDrawingBuffer={false}
    preventStyleDiffing={false}
    reuseMaps={false}
    scrollZoom={true}
    style={null}
    touchAction="none"
    touchRotate={false}
    touchZoom={true}
    transitionDuration={0}
    transitionEasing={[Function]}
    transitionInterpolator={
      LinearInterpolator {
        "propNames": Array [
          "longitude",
          "latitude",
          "zoom",
          "bearing",
          "pitch",
        ],
      }
    }
    transitionInterruption={1}
    visibilityConstraints={
      Object {
        "maxPitch": 85,
        "maxZoom": 24,
        "minPitch": 0,
        "minZoom": 0,
      }
    }
    visible={true}
    width="100%"
    zoom={11}
  />
</div>
`;
